app/models/ability.rb

```diff
@@ -0,0 +1,45 @@
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    user ||= User.new # guest user (not logged in)
+    # a signed-in user can do everything
+    if user.has_role? :admin
+      # an admin can do everything
+      can :manage, :all
+    else
+      can :read, [Article,Comment]
+      can :create, Comment
+
+      if user.has_role? :blogger
+        can :create, Article
+        can :update, Article do |article|
+          article.try(:user) == user
+        end
+        can :destroy, Article do |article|
+          article.try(:user) == user
+        end
+        can :manage, Comment do |comment|
+          comment.try(:article).try(:user).eql?(user)
```

Comment:
yes blogger is able to destroy any comment on his article
& normal user is viewer who has read access to article and can create comment